lib/internal/modules/cjs/loader.js

```diff
@@ -980,16 +992,14 @@ Module._resolveFilename = function(request, parent, isMain, options) {
     paths = Module._resolveLookupPaths(request, parent);
   }
 
-  if (parent && parent.filename) {
-    const filename = trySelf(parent.filename, request);
-    if (filename) {
-      const cacheKey = request + '\x00' +
-          (paths.length === 1 ? paths[0] : paths.join('\x00'));
-      Module._pathCache[cacheKey] = filename;
-      return filename;
-    }
+  // Try module self resoultion first
+  const selfResolved = trySelf(parent, request);
+  if (selfResolved) {
+    const cacheKey = request + '\x00' +
+         (paths.length === 1 ? paths[0] : paths.join('\x00'));
```

Comment:
```suggestion
         (paths.length === 1 ? paths[0] : ArrayPrototypeJoin(paths, '\x00'));
```

Reply:
I am sure this is a stupid question, why would it be better, in this case, using `Array.prototype.join`?

Reply:
Because if you import `ArrayPrototypeJoin` from primordials, then `delete Array.prototype.join` in user code won't cause the commonJS resolver to crash. node should be robust against user code modifying the environment.

Reply:
Wow, thanks for the detailed explanation, I wasn't thinking about that!